runelite-client/src/main/java/net/runelite/client/plugins/statusbars/BarRenderer.java

```diff
@@ -0,0 +1,170 @@
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.SKILL_ICON_HEIGHT;
+import static net.runelite.client.plugins.statusbars.StatusBarsOverlay.WIDTH;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.components.TextComponent;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+class BarRenderer
+{
+	private static final TextComponent textComponent = new TextComponent();
+
+	private final StatusBarsConfig config;
+	private final Supplier<Integer> maxValueSupplier;
+	private final Supplier<Integer> currentValueSupplier;
+	private final Supplier<Integer> healSupplier;
+	private final Supplier<Color> colorSupplier;
+	private final Supplier<Color> healColorSupplier;
+	private final Supplier<Image> iconSupplier;
+
+	private int maxValue;
+	private int currentValue;
+
+	private void refreshSkills()
+	{
+		maxValue = maxValueSupplier.get();
+		currentValue = currentValueSupplier.get();
+	}
+
+	private void renderBar(Graphics2D graphics, int x, int y, int height)
+	{
+		final Color fill = colorSupplier.get();
+
+		graphics.setColor(BACKGROUND);
+		graphics.drawRect(x, y, WIDTH - PADDING, height - PADDING);
+		graphics.fillRect(x, y, WIDTH, height);
+
+		final int filledHeight = getBarHeight(maxValue, currentValue, height);
+		graphics.setColor(fill);
+		graphics.fillRect(x + PADDING,
+				y + PADDING + (height - filledHeight),
+				WIDTH - PADDING * OFFSET,
+				filledHeight - PADDING * OFFSET);
+	}
+
+	private void renderIconsAndCounters(Graphics2D graphics, int x, int y)
+	{
+		final String counterText = Integer.toString(currentValue);
+		final int widthOfCounter = graphics.getFontMetrics().stringWidth(counterText);
+		final int centerText = (WIDTH - PADDING) / 2 - (widthOfCounter / 2);
+
+		if (config.enableCounter())
+		{
+			graphics.setFont(FontManager.getRunescapeSmallFont());
+			textComponent.setColor(Color.WHITE);
+			textComponent.setText(counterText);
+			textComponent.setPosition(new java.awt.Point(x + centerText + 1, y + COUNTER_ICON_HEIGHT));
+		}
+		else
+		{
+			textComponent.setText("");
+		}
+
+		if (config.enableSkillIcon())
+		{
+			final Image icon = iconSupplier.get();
+			graphics.drawImage(icon, x + ICON_AND_COUNTER_OFFSET_X + PADDING, y + ICON_AND_COUNTER_OFFSET_Y - icon.getWidth(null), null);
+			textComponent.setPosition(new java.awt.Point(x + centerText + 1, y + SKILL_ICON_HEIGHT));
+		}
+
+		textComponent.render(graphics);
+	}
+
+	private void renderRestore(Graphics2D graphics, int x, int y, int height)
+	{
+		final Color color = healColorSupplier.get();
+		final int heal = healSupplier.get();
+
+		if (heal <= 0)
+		{
+			return;
+		}
+
+		final int filledCurrentHeight = getBarHeight(maxValue, currentValue, height);
+		int filledHeight = getBarHeight(maxValue, heal, height);
+		graphics.setColor(color);
+
+		if (filledHeight + filledCurrentHeight > height)
```

Comment:
Does this branch of the `if` statement only draw the restoration bar? It seems like overheal should be drawn so that both current -> maximum and minimum -> final overheal value are drawn on.

That is, getting the first of these images, not the second. (run this only if the overheal exceeds maximum, but run the `else` branch of code regardless)

![full-overheal](https://user-images.githubusercontent.com/2199511/52457869-ea9dc980-2b53-11e9-9155-16a7c4a1d11f.png) ![only-overheal](https://user-images.githubusercontent.com/2199511/52457875-f6898b80-2b53-11e9-89e9-3c2ac40cb69d.png)

Reply:
This branch only draws the restorationbar yes, i'm not entirely sure what you mean with that both should be drawn on? 
It just checks if its more than the maximum value, and if it is then it draws to full hp bar + color change do indicate overhealing which should be fine?